.github/workflows/sda-pipeline.yml

```diff
@@ -0,0 +1,119 @@
+      if: failure()
+      run: kubectl logs postgres-sda-db-0
+    - name: Deploy mock oidc server
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/deploy-oidc.sh
+    - name: Deploy minio
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/deploy-minio.sh
+    - name: Wait for minio to become ready
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/wait-for-pods.sh minio
+    - name: debug minio
+      if: failure() && matrix.test == 's3-inbox'
+      run: kubectl describe pod -l app=minio
+    - name: log minio
+      if: failure() && matrix.test == 's3-inbox'
+      run: kubectl logs $(kubectl get pods | grep minio | awk '{print $1}')
+    - name: Start CEGA services
+      run: bash ./dev_tools/scripts/deploy-cega.sh
+    - name: Wait for CEGA to become ready
+      run: bash ./dev_tools/scripts/wait-for-pods.sh cega-mq
+    - name: debug cega
+      if: failure()
+      run: |
+        for c in cega-users cega-mq
+        do kubectl describe pod -l app=$c
+        done
+    - name: log cega
+      if: failure()
+      run: |
+        for c in cega-users cega-mq
+        do kubectl logs $(kubectl get pods | grep $c | awk '{print $1}')
+        done
+    - name: Deploy SDA message broker
+      run: bash ./dev_tools/scripts/deploy-mq.sh
+    - name: Wait for broker to become ready
+      run: bash ./dev_tools/scripts/wait-for-pods.sh broker
+    - name: debug MQ
+      if: failure()
+      run: kubectl describe pod -l role=broker
+    - name: log MQ
+      if: failure()
+      run: kubectl logs broker-sda-mq-0
+    - name: Deploy the SDA stack for posix
+      if: matrix.test == 'sftp-inbox'
+      run: |
+        kubectl apply -f .github/ci_tests/posix-volumes.yaml;
+        bash ./dev_tools/scripts/deploy-sda-posix.sh;
+    - name: Wait for posix SDA stack to become ready
+      if: matrix.test == 'sftp-inbox'
+      run: bash ./dev_tools/scripts/wait-for-pods.sh "${{ env.posix_svc_list }}"
+    - name: debug posix sda
+      if: failure() && matrix.test == 'sftp-inbox'
+      run: |
+        for s in ${{ env.posix_svc_list }}
+        do kubectl describe pod -l role=$s
+        done
+    - name: log posix sda
+      if: failure() && matrix.test == 'sftp-inbox'
+      run: |
+        for s in ${{ env.posix_svc_list }}
+        do kubectl logs $(kubectl get pods | grep $s | awk '{print $1}')
+        done
+    - name: Deploy the SDA stack for s3
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/deploy-sda-s3.sh
+    - name: Wait for s3 SDA stack to become ready
+      if: matrix.test == 's3-inbox'
+      run: bash ./dev_tools/scripts/wait-for-pods.sh "${{ env.s3_svc_list }}"
+    - name: debug s3 sda
+      if: failure() && matrix.test == 's3-inbox'
+      run: |
+        for s in ${{ env.s3_svc_list }}
+        do kubectl describe pod -l role=$s
+        done
+    - name: log s3 sda
+      if: failure() && matrix.test == 's3-inbox'
```

Comment:
This can be simplified without any `if` statements. i.e you have abase list of services and just add `sftp-inbox` or `s3-inbox` depending on which matrix that are being run.

Reply:
That sounds like an if statement to me. At least as long as we run them on the same workflow action.

Reply:
You have duplicate blocks like this, which could be just one.
Here you have env.posix_svc_list and env.s3_svc_list

```
   - name: debug posix sda
      if: failure() && matrix.test == 'sftp-inbox'
      run: |
        for s in ${{ env.posix_svc_list }}
        do kubectl describe pod -l role=$s
        done
    - name: debug s3 sda
      if: failure() && matrix.test == 's3-inbox'
      run: |
        for s in ${{ env.s3_svc_list }}
        do kubectl describe pod -l role=$s
        done
```

You can have just one list env.svc_list and depending on is the run is s3 or posix that specific inbox is added to the list using 
```
echo "s3-inbox" >> env.svc_list
```

Reply:
Does that persist across different steps, or do you need to add it every single time? Because if so, then we need to do it on log, debug and wait.

Reply:
If you save the list as a file, yes.
And the only one that differs is `auth`

Reply:
I'm not sure I follow the idea demoed above, but one option would possibly be to have this in shell logic early on with https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#environment-files, e.g. (untested)

```
- name: fixup env
  run: |
      if [ "${{ matrix.test }}" = "sftp-inbox" ]; then
       echo "SVCS=backup doa finalize inbox ingest intercept mapper verify" >> "$GITHUB_ENV"; 
     else
       echo "SVCS=auth backup doa finalize inbox ingest intercept mapper verify" >> "$GITHUB_ENV"; 
     fi
```

Reply:
Good idea. Will test this!